docker-run-opencv-ovms.sh

```diff
@@ -0,0 +1,226 @@
+	TAG=$CONTAINER_IMAGE_OVERRIDE
+fi
+
+cids=$(docker ps  --filter="name=ovms-client" -q -a)
+cid_count=`echo "$cids" | wc -w`
+SERVER_CONTAINER_NAME="model-server"
+CLIENT_CONTAINER_NAME="ovms-client"$(($cid_count))
+LOG_FILE_NAME="ovms-client"$(($cid_count))".log"
+
+#echo "barcode_disabled: $BARCODE_DISABLED, barcode_interval: $BARCODE_INTERVAL, ocr_interval: $OCR_INTERVAL, ocr_device: $OCR_DEVICE, ocr_disabled=$OCR_DISABLED, class_disabled=$CLASSIFICATION_DIABLED"
+pre_process=""
+if grep -q "rtsp" <<< "$INPUTSRC"; then
+	# rtsp
+	# todo pass depay info
+	inputsrc=$INPUTSRC" ! rtph264depay "
+	INPUTSRC_TYPE="RTSP"
+	pre_process="pre-process-backend=vaapi-surface-sharing -e pre-process-config=VAAPI_FAST_SCALE_LOAD_FACTOR=1"
+
+
+elif grep -q "file" <<< "$INPUTSRC"; then
+	# filesrc	
+	arrfilesrc=(${INPUTSRC//:/ })
+	# use vids since container maps a volume to this location based on sample-media folder
+	# TODO: need to pass demux/codec info
+	inputsrc="filesrc location=vids/"${arrfilesrc[1]}" ! qtdemux ! h264parse "
+	INPUTSRC_TYPE="FILE"
+	decode_type="vaapidecodebin"
+	pre_process="pre-process-backend=vaapi-surface-sharing -e pre-process-config=VAAPI_FAST_SCALE_LOAD_FACTOR=1"
+
+elif grep -q "video" <<< "$INPUTSRC"; then
+	# v4l2src /dev/video*
+	# TODO need to pass stream info
+	inputsrc="v4l2src device="$INPUTSRC
+	INPUTSRC_TYPE="USB"
+	TARGET_USB_DEVICE="--device=$INPUTSRC"
+	decode_type="videoconvert ! video/x-raw,format=BGR"
+	pre_process=""
+
+else
+	# rs-serial realsenssrc
+	# TODO need to pass depthalign info
+	inputsrc="realsensesrc cam-serial-number="$INPUTSRC" stream-type=0 align=0 imu_on=false"
+    # add realsense color related properties if any
+	if [ "$COLOR_WIDTH" != 0 ]; then
+		inputsrc=$inputsrc" color-width="$COLOR_WIDTH
+	fi
+	if [ "$COLOR_HEIGHT" != 0 ]; then
+		inputsrc=$inputsrc" color-height="$COLOR_HEIGHT
+	fi
+	if [ "$COLOR_FRAMERATE" != 0 ]; then
+		inputsrc=$inputsrc" color-framerate="$COLOR_FRAMERATE
+	fi
+	INPUTSRC_TYPE="REALSENSE"
+	decode_type="decodebin ! videoconvert ! video/x-raw,format=BGR"
+	pre_process=""
+	cameras=`ls /dev/vid* | while read line; do echo "--device=$line"; done`
+	TARGET_GPU_DEVICE=$TARGET_GPU_DEVICE" "$cameras	
+fi
+
+#todo: this will need to be updated to how we are supporting these configs
+if [ "${OCR_DISABLED}" == "0" ] && [ "${BARCODE_DISABLED}" == "0" ] && [ "${CLASSIFICATION_DISABLED}" == "0" ] && [ "${REALSENSE_ENABLED}" == "0" ]; then
+	pipeline="yolov5s_full.sh"
+	
+elif [ "${OCR_DISABLED}" == "1" ] && [ "${BARCODE_DISABLED}" == "1" ] && [ "${CLASSIFICATION_DISABLED}" == "1" ]; then
+	pipeline="yolov5s.sh"
+elif [ "${OCR_DISABLED}" == "1" ] && [ "${BARCODE_DISABLED}" == "1" ] && [ "${CLASSIFICATION_DISABLED}" == "0" ]; then
+	pipeline="yolov5s_effnetb0.sh"
+elif [ "${REALSENSE_ENABLED}" == "1" ]; then
+	# TODO: this will not work for diff pipelines like _full and _effnetb0 etc
+	pipeline="yolov5s_realsense.sh"
+	
+else
+	echo "Not implemented"
+	exit 0
```

Comment:
if not support, should we do exit 1?

Reply:
[based on this common used status code](https://www.adminschoice.com/exit-error-codes-in-bash-and-linux-os), i would suggest using exit code 2 or 127

Reply:
Modified to exit 2